onnxruntime/core/providers/coreml/builders/impl/depthtospace_op_builder.cc

```diff
@@ -145,6 +145,23 @@ bool DepthToSpaceOpBuilder::IsOpSupportedImpl(const Node& node, const OpBuilderI
       LOGS(logger, VERBOSE) << "DepthToSpace: CRD mode requires static shape";
       return false;
     }
+
+    if (mode == "DCR" && input_params.coreml_version < 7) {
+      int32_t input_type = ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED;
+      if (!GetType(*input_defs[0], input_type, logger)) {
+        LOGS(logger, VERBOSE) << "[" << node.OpType() << "] Get Input type failed";
+        return false;
+      }
```

Comment:
nit: this is checked in the base OpBuilder prior to calling IsOpSupportedImpl

https://github.com/microsoft/onnxruntime/blob/e5c2e5084948cacc1d58d6d9b38ae4da85d4ce92/onnxruntime/core/providers/coreml/builders/impl/base_op_builder.cc#L134